src/components/Views/ObservationEditor/ObservationEditor.js

```diff
@@ -496,6 +516,76 @@ class ObservationEditor extends React.Component<
     }
   };
 
+  observationExists = () => {
+    const { observations, selectedObservation } = this.props;
+
+    if (selectedObservation && observations) {
+      if (observations.find(o => o.id === selectedObservation.id)) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  allowEditing = (index: number) => {
+    const allowEdit =
+      !this.observationExists() ||
+      (this.observationExists() && index >= this.state.numExistingPhotos);
+
+    return allowEdit;
+  };
+
+  hasEdits = () => {
+    const { selectedObservation } = this.props;
+    const {
+      numExistingPhotos,
+      existingNotes,
+      numExistingFieldsAnswered,
+      text
+    } = this.state;
+
+    if (selectedObservation) {
+      if (this.observationExists()) {
+        const currentFieldsAnswered = selectedObservation.fields.filter(
+          field => field.answered
+        ).length;
+
+        const hasEdits =
+          selectedObservation.media.length !== numExistingPhotos ||
+          text !== existingNotes ||
+          currentFieldsAnswered !== numExistingFieldsAnswered;
+
+        return hasEdits;
+      }
+      const hasFieldAnswered = !!selectedObservation.fields.find(
+        field => field.answered
+      );
+      const noEdits =
+        selectedObservation &&
+        selectedObservation.media.length === 0 &&
+        selectedObservation.notes === '' &&
+        !hasFieldAnswered;
+
+      return !noEdits;
+    }
+
+    return false;
+  };
+
+  handleCancel = () => {
+    const { navigation, observationSource, showCancelModal } = this.props;
+    if (this.hasEdits()) {
+      showCancelModal();
+    } else if (observationSource === 'map') {
+      navigation.navigate({
```

Comment:
Won't this mess up the back button on the android?

Reply:
Hmm how so?

Reply:
Oh never mind, I thought this was the cancel to go back button. This should be fine

Reply:
👍